frontend/tests/components/applyForm/validate.test.ts

```diff
@@ -0,0 +1,92 @@
+import { RJSFSchema } from "@rjsf/utils";
+
+import { UiSchema } from "src/components/applyForm/types";
+import {
+  validateFormData,
+  validateFormSchema,
+  validateUiSchema,
+} from "src/components/applyForm/validate";
+
+describe("validateFormSchema", () => {
+  it("should validate a correct form schema", () => {
+    const validSchema: RJSFSchema = {
+      title: "test schema",
+      properties: {
+        name: { type: "string", title: "test name" },
+      },
+      required: ["name"],
+    };
+
+    expect(() => validateFormSchema(validSchema)).not.toThrow();
+  });
+
+  it("should throw an error for an invalid form schema", () => {
+    const invalidSchema: RJSFSchema = {
+      title: "test schema",
+      properties: {
+        name: { type: "invalid type" as "string", title: "test name" },
```

Comment:
[rant] this is the sort of thing that annoys me about typescript in unit tests - if you're explicitly trying to test something that is outside of the typing you have to do something like this, which ends up looking dumb and being kind of pointless from the perspective of typechecking

Reply:
Types are for more than just application code https://www.youtube.com/watch?v=0mCsluv5FXA